samza-core/src/main/java/org/apache/samza/zk/ZkMetadataStoreFactory.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.zk;
+
+import org.apache.samza.config.Config;
+import org.apache.samza.config.ZkConfig;
+import org.apache.samza.coordinator.stream.messages.SetContainerHostMapping;
+import org.apache.samza.metadatastore.MetadataStore;
+import org.apache.samza.metadatastore.MetadataStoreFactory;
+import org.apache.samza.metrics.MetricsRegistry;
+import org.apache.samza.SamzaException;
+
+/**
+ * Builds the {@link ZkMetadataStore} based upon the provided {@link Config}
+ * and {@link MetricsRegistry}.
+ */
+public class ZkMetadataStoreFactory implements MetadataStoreFactory {
+
+  private static final String TASK_LOCALITY_NAMESPACE = "TaskLocalityNamespace";
```

Comment:
Where is this used other than in the unit test? 
https://github.com/apache/samza/blob/029f652cb0b0bb597c469ec8d748a4120294f3dc/samza-core/src/test/java/org/apache/samza/zk/TestZkMetadataStore.java#L39

Reply:
It will be used in the follow up patch where after the stream processors group agree on a JobModel, each one of them will update the taskLocality before launching the `SamzaContainer`.

Reply:
This doesn't belong in the Store Factory implementation. See comment about namespace selection below. Also, namespace shouldn't be specific to implementation, this should be the same whether for coordinator stream and ZK.

Reply:
Will change.